core/services/vrf/v2/listener_v2_log_listener.go

```diff
@@ -0,0 +1,438 @@
+package v2
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/services/vrf/vrfcommon"
+	"github.com/smartcontractkit/chainlink/v2/core/utils/mathutil"
+)
+
+func (lsn *listenerV2) runLogListener(
+	pollPeriod time.Duration,
+	minConfs uint32,
+) {
+	lsn.l.Infow("Listening for run requests via log poller",
+		"minConfs", minConfs)
+	ticker := time.NewTicker(pollPeriod / 2)
+	defer ticker.Stop()
+	var (
+		lastProcessedBlock int64
+		startingUp         = true
+	)
+	ctx, cancel := lsn.chStop.NewCtx()
+	defer cancel()
+	for {
+		select {
+		case <-lsn.chStop:
+			return
+		case <-ticker.C:
+			start := time.Now()
+			lsn.l.Debugw("log listener loop")
+			// Filter registration is idempotent, so we can just call it every time
+			// and retry on errors using the ticker.
+			err := lsn.chain.LogPoller().RegisterFilter(logpoller.Filter{
```

Comment:
I wonder if having multiple jobs frequently re-register on the same LP instance can cause any issues. Register may not be idempotent if name changes?

Reply:
never mind. filter does not become "dirty" if it's an existing filter